File: Backend/PFE-back-master/backendPfe-master/src/main/java/com/example/backend_pfe/entities/Notification.java
package com.example.backend_pfe.entities;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Map;


import org.springframework.data.mongodb.core.mapping.Document;

import jakarta.persistence.Entity;
import jakarta.persistence.*;
import jakarta.persistence.GenerationType;
import lombok.*;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Notification {
	
	@Id
	@GeneratedValue (strategy=GenerationType.IDENTITY) 
    private Long id;
    private long alertId;
    private String username;
    private long userId;
    private String message;
    private boolean isSent;
    private LocalDateTime createdAt;
    private boolean viewed;
    private String tickerDetails;
    
    
    private String alertName;      
    private String alertType;      
    private String alertFieldName;
    private String alertoperator;
    private double alertthreshold;
}

File: Mobile Front/PFE-Mobile-Final-master/lib/entities/notification.dart
class UserNotification {
  final int id;
  final int alertId;
  final String username;
  final int userId;
  final String message;
  final bool isSent;
  final String createdAt;
  final bool viewed;
  final String tickerDetails;

  final String alertName;     
  final String alertType;   
  final String alertFieldName;
  final String alertoperator;
  final double alertthreshold;


  UserNotification({
    required this.id,
    required this.alertId,
    required this.username,
    required this.userId,
    required this.message,
    required this.isSent,
    required this.createdAt,
    required this.viewed,
    required this.tickerDetails,

    required this.alertName,
    required this.alertFieldName,
    required this.alertType,
    required this.alertthreshold,
    required this.alertoperator,
    

  });

  factory UserNotification.fromJson(Map<String, dynamic> json) {
    return UserNotification(
      id: json['id'],
      alertId: json['alertId'],
      username: json['username'],
      userId: json['userId'],
      message: json['message'],
      isSent: json['isSent'] ?? false, 
      createdAt: json['createdAt'],
      viewed: json['viewed'] ?? false, 
      tickerDetails: json['tickerDetails'],

      alertName: json['alertName'],
      alertFieldName: json['alertFieldName'],
      alertType: json['alertType'],
      alertthreshold: json['alertthreshold'],
      alertoperator: json['alertoperator'],
    
    );
  }
}
